### DISCUSSION TITLE
When trying to give an Option with a prompt value, keep asking for a value.

### DISCUSSION THREAD

**Discussion by afcabezasq** [fastapi/typer#603] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the Typer documentation, with the integrated search.
- [X] I already searched in Google "How to X in Typer" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to Typer but to [Click](https://github.com/pallets/click).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
def generateConfigFile(server:str, database:str, user:str, password:str):
    
    configurations: dict[str,str] = {
        'Server': server,
        'Database': database,
        'User': user,
        'Password': password
    }

    with open ('config.yaml','wt',encoding="utf-8") as file:
        yaml.dump(configurations,file)
  
@app.command()
def config(server:str = typer.Option(..., prompt_required=True, prompt=True),
           database:str = typer.Option(..., prompt_required=True, prompt=True),
           user: str = typer.Option(..., prompt_required=False, prompt=True),
           password: str =  typer.Option(..., prompt_required=False, prompt=True, hide_input=True)):
    """
    Run this command to install the database configurations
    """
    with Progress(
        SpinnerColumn(),
        TextColumn("[progress.description]{task.description}"),
        transient=True,
    ) as progress:
        progress.add_task(description="Processing...", total=None)
        progress.add_task(description="Preparing...", total=None)
        
        generateConfigFile(server,database,user,password)
        print("Done!")
```

### Description

I have a similar code as above that generates a configuration file, but I would like to have the option to have empty user and empty password. When I run my application:
`
app config

Server: myServer
Database: myDatabase
User:
User:
User:
`

If I want to have the option empty, the application keep asking for a parameter, but I would like the application to recognize the empty string, or what workaround can I use for that? I wouldn't want to use the input function, just want to know if there is a typer option to do that

### Operating System

Windows

### Typer Version

0.7.0

### Python Version

3.9.13

**Comment by afcabezasq** [comment, marked as answer]
Neverminded I figured it out 🙈, I just set the following parameters typer.Option(default="", show_default=False, prompt=True, hide_input=True)).

**Reply by timonrieger** [reply]
Thanks, you helped me out 🙏 
~`show_default=False` is not needed btw as empty strings are falsy already 😄~, it actually is, otherwise would render `API Key []: ` in the prompt
